src/main/Application.h

```diff
@@ -257,6 +262,19 @@ class Application
     // copy made of `cfg`.
     static pointer create(VirtualClock& clock, Config const& cfg,
                           bool newDB = true);
+    template <typename T>
+    static pointer create(VirtualClock& clock, Config const& cfg,
```

Comment:
if you were returning std::shared_ptr<T> instead of pointer, it would save you a bunch of work elsewhere

Reply:
Github is removing <> from ::shared_ptr&lt;T> in comments.